spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/EnableRabbitIntegrationTests.java

```diff
@@ -1495,6 +1545,20 @@ public DirectRabbitListenerContainerFactory directListenerContainerFactory() {
 			return factory;
 		}
 
+		@Bean
+		public SimpleRabbitListenerContainerFactory consumerBatchContainerFactory() {
+			SimpleRabbitListenerContainerFactory factory = new SimpleRabbitListenerContainerFactory();
+			factory.setConnectionFactory(rabbitConnectionFactory());
+			factory.setConsumerTagStrategy(consumerTagStrategy());
+			factory.setBatchListener(true);
+			factory.setContainerConfigurer(container -> {
+				container.setConsumerBatchEnabled(true);
+				container.setDeBatchingEnabled(true);
```

Comment:
Why these two options can't be populated directly by the `SimpleRabbitListenerContainerFactory`  when that `etBatchListener(true)` is used?

Reply:
Because the original purpose of `batchListener` was for producer batches.

Reply:
M-m-m. Can't we cover those "producer batches" with a condition based on the `BatchingStrategy` provided?